sdk/python/lib/pulumi/output.py

```diff
@@ -253,27 +253,32 @@ def from_input(val: Input[T]) -> 'Output[T]':
         if isinstance(val, dict):
             # Since Output.all works on lists early, serialize this dictionary into a list of lists first.
             # Once we have a output of the list of properties, we can use an apply to re-hydrate it back into a dict.
-            transformed_items = [[k, Output.from_input(v)] for k, v in val.items()]
-            return Output.all(*transformed_items).apply(lambda props: {k: v for k, v in props}, True)
+            dict_items = [[k, Output.from_input(v)] for k, v in val.items()]
+            # type checker doesn't like returing a Dict in the apply callback
+            fn = cast(Callable[[List[Any]], T], lambda props: {k: v for k, v in props}) # pylint: disable=unnecessary-comprehension
+            return Output.all(*dict_items).apply(fn, True)
 
         if isinstance(val, list):
-            transformed_items = [Output.from_input(v) for v in val]
-            return Output.all(*transformed_items)
+            list_items: List[Union[Any, Awaitable[Any], Output[Any]]] = [Output.from_input(v) for v in val]
+            # type checker doesn't like returning a list
+            output: Output[T] = cast(Output[T], Output.all(list_items))
+            return output
 
         # If it's not an output, list, or dict, it must be known and not secret
-        is_known_fut = asyncio.Future()
-        is_secret_fut = asyncio.Future()
+        is_known_fut: asyncio.Future = asyncio.Future()
```

Comment:
Totally curious - were these necessary?  I would have expected these to be the default inferred types?

Reply:
Strange right? Without it, we get:`lib/pulumi/output.py:268: error: Need type annotation for 'is_known_fut'`

Reply:
Looks like it's because `Future` is generic, and this error message is trying to tell you you need a type argument, even though `asyncio.Future` implicitly means `asyncio.Future[Any]`.  We should probably provide an actual type argument in our annotation.

https://github.com/python/mypy/issues/4052

Reply:
Made this change where applicable. In a few instances, we are resolving futures with `Any` values.